cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/BatchTransformExecutor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch.mapreduce;
+
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.batch.BatchTransformDetail;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Executes chain of transforms
+ * @param <IN> Type of input
+ */
+public class BatchTransformExecutor<IN> {
```

Comment:
should be Destroyable, and you should destroy all the transforms after it is done.